pennylane/templates/tensornetworks/mps.py

```diff
@@ -113,39 +125,65 @@ def circuit(template_weights):
         2: ───────────────╰X──RY(-0.30)─╭●──RY(0.10)──┤
         3: ─────────────────────────────╰X──RY(-0.30)─┤  <Z>
 
-    """
+        MPS can also be used with an ``offset`` argument that shifts the positioning the subsequent blocks relative to the default ``n_block_wires/2``.
+
+        .. code-block:: python
+
+            import pennylane as qml
+            import numpy as np
+
+            def block(weights, wires):
+                qml.MultiControlledX(wires=[wires[i] for i in range(len(wires))])
+
+            n_wires = 8
+            n_block_wires = 4
+            n_params_block = 2
+
+            dev= qml.device('default.qubit',wires=n_wires)
+            @qml.qnode(dev)
+            def circuit():
+                qml.MPS(range(n_wires),n_block_wires, block, n_params_block, offset = -1)
+                return qml.state()
+
+        >>> print(qml.draw(circuit, expansion_strategy='device')())
+        0: ─╭●─────────────┤  State
+        1: ─├●─╭●──────────┤  State
+        2: ─├●─├●─╭●───────┤  State
+        3: ─╰X─├●─├●─╭●────┤  State
+        4: ────╰X─├●─├●─╭●─┤  State
+        5: ───────╰X─├●─├●─┤  State
+        6: ──────────╰X─├●─┤  State
+        7: ─────────────╰X─┤  State
 
-    num_params = 1
-    """int: Number of trainable parameters that the operator depends on."""
+    """
 
     num_wires = AnyWires
     par_domain = "A"
 
     @classmethod
     def _unflatten(cls, data, metadata):
         new_op = cls.__new__(cls)
-        new_op._hyperparameters = dict(metadata[1])
-        Operation.__init__(new_op, data, wires=metadata[0])
+        new_op._hyperparameters = dict(metadata[1])  # pylint: disable=protected-access
+        new_op._weights = data[0] if len(data) else None  # pylint: disable=protected-access
+        Operation.__init__(new_op, *data, wires=metadata[0])
         return new_op
 
     def __init__(
         self,
         wires,
         n_block_wires,
         block,
-        n_params_block,
+        n_params_block=0,
         template_weights=None,
+        offset=0,
         id=None,
+        **kwargs,
     ):
-        ind_gates = compute_indices_MPS(wires, n_block_wires)
-        n_wires = len(wires)
-        n_blocks = int(n_wires / (n_block_wires / 2) - 1)
-
-        if template_weights is None:
-            template_weights = np.random.rand(n_params_block, int(n_blocks))
+        ind_gates = compute_indices_MPS(wires, n_block_wires, offset)
+        n_blocks = self.get_n_blocks(wires, n_block_wires, offset)
 
-        else:
-            shape = qml.math.shape(template_weights)[-4:]  # (n_params_block, n_blocks)
+        if template_weights is not None:
+            shape = qml.math.shape(template_weights)[-4:]  # (n_blocks, n_params_block)
```

Comment:
Did we ever figure out what this magic [-4:] does? :mage_man: :crystal_ball: :sparkles:

Reply:
I don't think it is necessary, I am removing it.